### PR TITLE
test: AWS S3 integration test

### PR DESCRIPTION
**Pull request by zucchinidev**
[#182285734](https://www.pivotaltracker.com/story/show/182285734)

### Checklist:

* ~~[ ] Have you added Draft Release Notes in `docs/draft-release-notes.md`?~~
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: aws-s3-bucket.yml [hunk 1]
```diff
@@ -20,26 +20,26 @@ display_name: AWS S3 Bucket
...
-  display_name: 'Private'
-  properties:
-    acl: private
-- name: public-read
-  id: 04317eaa-11ac-4c5f-b77f-eb005fe977fe
-  description: 'Public-read S3 bucket'
-  display_name: 'Public Read'
-  properties:
-    acl: public-read
+  - name: private
```
Comment (FelisiaM) [inline aws-s3-bucket.yml:26]: There seem to be a lot of formatting changes that make this yaml file not consistent with the others. If you used goland I think the go formatted makes this change on save - if that's the case you can exclude .yml files from the formatting rules. I would suggest you revert the formatting to what it was to keep it consistent with other files and to keep the changes in this PR to a min.

File: integration-tests/s3_test.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+var customS3Plan = map[string]any{
+	"name":        "custom-plan",
+	"id":          "9dfa265e-1c4d-40c6-ade6-b341ffd6ccc3",
+	"description": "custom S3 plan defined by customer",
+	"acl":         "private",
+	"metadata": map[string]any{
+		"displayName": "custom S3 service",
+	},
+	"labels": map[string]any{
```
Comment (FelisiaM) [inline integration-tests/s3_test.go:24]: labels are not a parameter that can be defined for a plan

**Review by FelisiaM — CHANGES_REQUESTED** [review]

File: aws-s3-bucket.yml (hunk 1 above)
Comment (zucchinidev) [reply aws-s3-bucket.yml]: Done!! Thank you very much!

File: integration-tests/s3_test.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+		)
+	})
+
+	Describe("updating instance", func() {
+		var instanceID string
+
+		BeforeEach(func() {
+			instanceID, _ = broker.Provision(s3ServiceName, customS3Plan["name"].(string), nil)
+
+			Expect(mockTerraform.FirstTerraformInvocationVars()).To(
```
[the remark this answers is not in the document]
Comment (zucchinidev) [reply integration-tests/s3_test.go:111]: Very good question @FelisiaM, I have followed the same pattern used in gcp, I do not say it is the right thing, it is only a clarification. What I do see that could be an advantage of this test, is that it is tested on the custom plan, which is a very limited plan, therefore it is proven that the values are allocated correctly. Does it make sense? You want me to change it?

File: integration-tests/s3_test.go (hunk 2 above)
Comment (zucchinidev) [reply integration-tests/s3_test.go]: Done!! Thank you very much!!

File: integration-tests/s3_test.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+			)
+		})
+
+		It("should not allow changing of plan defined properties", func() {
+			_, err := broker.Provision(s3ServiceName, "private", map[string]any{"acl": "public-read"})
+
+			Expect(err).To(MatchError(ContainSubstring("plan defined properties cannot be changed: acl")))
+		})
+
+		DescribeTable("property constraints",
```
[the remark this answers is not in the document]
Comment (zucchinidev) [reply integration-tests/s3_test.go]: Changed! I was thinking about future properties that I suppose will be added in the next few weeks, but better not to take anything for granted and change it when appropriate. Thank you very much!!

File: integration-tests/s3_test.go [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			bindResult, err := broker.Bind(s3ServiceName, customS3Plan["name"].(string), instanceID, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(bindResult).To(
+				Equal(map[string]any{
+					"access_key_id":     "subsequent.access.key.id.test",
+					"secret_access_key": "initial.secret.access.key.test",
```
[the remark this answers is not in the document]
Comment (zucchinidev) [reply integration-tests/s3_test.go]: Added and changed. Thank you very much!

File: integration-tests/s3_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (FelisiaM) [reply integration-tests/s3_test.go:111]: I would suggest that if we want to test the custom plan values those should be tested in a test that explains the intent. So a test about provisioning. As it stands this test which claims to be testing updates could be failing for provision related issues and that is confusing and expands the responsibilities of this test.

File: integration-tests/integration_test_suite_test.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package integration_tests_test
```
[the remark this answers is not in the document]
Comment (FelisiaM) [reply integration-tests/integration_test_suite_test.go:1]: I would suggest you still change to `integration_test` just to be consistent with the other GCP tests

**Comment by FelisiaM** [comment]
Since these are the first integration tests that we are adding in AWS there needs to be a way to run them in Github actions and in CI. 
For this, you first add new options to the `Makefile` file (like `run-tests`, `run-integration-tests` in GCP).
To run them in the Github Actions follow the same pattern as in GCP by adding a workflow: https://github.com/cloudfoundry/csb-brokerpak-gcp/blob/main/.github/workflows/run-tests.yml 

I think CI needs a bit of pairing to get your head around Concourse, but feel free to see examples

File: integration-tests/integration_test_suite_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (zucchinidev) [reply integration-tests/integration_test_suite_test.go:1]: Of course!! Thank you!

File: integration-tests/s3_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zucchinidev) [reply integration-tests/s3_test.go:111]: It makes totally sense! I will do it. Thank you very much

[the remark this answers is not in the document]
Comment (zucchinidev) [reply integration-tests/s3_test.go]: Thinking it over, it does not bring any value other than the previously provided, I will eliminate the expect

File: integration-tests/integration_test_suite_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (zucchinidev) [reply integration-tests/integration_test_suite_test.go]: Done!

**Review by jimbo459 — APPROVED** [review]

**Pull request merged**